mobilecoind/api/proto/mobilecoind_api.proto

```diff
@@ -611,6 +615,54 @@ message GenerateBurnRedemptionTxResponse {
     TxProposal tx_proposal = 1;
 }
 
+// Generate a simple swap proposal. The result is a signed contingent input
+// which trades one currency for another and is suitable for use with the deqs.
+// (This API is restrictive and doesn't let you build more complex SCIs.)
+message GenerateSwapRequest {
+    // Monitor id sending the funds.
+    bytes sender_monitor_id = 1;
+
+    // Subaddress to return change to.
+    uint64 change_subaddress = 2;
+
+    // A specific input, whose value will be offered in full by the swap.
+    //
+    // This becomes the "base token" amount from the point of view of a quoting service.
+    //
+    // You may need to conduct a self-spend first to make an input of exactly
+    // the correct value before using this API if none of your inputs match
+    // the volume of the quote you want to make.
+    UnspentTxOut input = 3;
+
+    // The u64 value we are asking for in exchange for our input
+    //
+    // This becomes the "counter token" value from the point of view of a quoting service.
+    uint64 counter_value = 4;
+
+    // The token_id we are asking for in exchange for our input
+    //
+    // This becomes the "counter token" token id from the point of view of a quoting service.
+    uint64 counter_token_id = 5;
+
+    // If set to false, the offer is "all or nothing", the entire counter token value must be supplied,
+    // in exchange for the entire value of the input we are signing over.
+    // Otherwise, it is a "partial-fill" SCI, and can be filled at less than the maximum volume for
+    // proportionally more of the input value.
+    bool allow_partial_fill = 6;
```

Comment:
Just a side note, the deqs currently only supports partial fill orders but it shouldn't be too difficult to change that. I think its still good to have this flag here.